apps/model/lib/model/route_pattern.ex

```diff
@@ -22,8 +23,9 @@ defmodule Model.RoutePattern do
           direction_id: Model.Direction.id(),
           name: String.t(),
           time_desc: String.t() | nil,
-          typicality: 0..4,
+          typicality: 0..5,
           sort_order: integer(),
-          representative_trip_id: Model.Trip.id()
+          representative_trip_id: Model.Trip.id(),
+          is_canonical: 0..2 | nil
```

Comment:
question: Why are we saying this could be possibly `nil`? I thought we were using `0` to represent `null` in JSON…am I missing something here?